internal/verifier/verifier.go

```diff
@@ -122,3 +128,47 @@ func (verifier *AmberProvenanceMetadataVerifier) Verify(provenanceFilePath strin
 
 	return nil
 }
+
+// Internal intermediate representation of Provenances for verification.
+// We use the ProvenanceIR to
+// (1) map different provenance formats, and
+// (2) hold reference values.
+// To be usable with different provenance formats, we allow fields to be empty ([]) and to hold several reference values.
+type ProvenanceIR struct {
+	BinarySHA256Digest []string
+}
+
+func (p *ProvenanceIR) FromSLSA(provenance *slsa.ValidatedProvenance) {
```

Comment:
I am not sure what golang conventions are, but I think if `FromSLSA` and `FromAmber` are functions, instead of methods on `ProvenanceIR` the code will be nicer. In particular, you'd have:

```
actual := verifier.FromSLSA(&provenances[index])
```

instead of 

```
var actual verifier.ProvenanceIR
actual.FromSLSA(&provenances[index])
```